service/errors.go

```diff
@@ -71,12 +71,16 @@ func (e *DomainError) Error() string {
 		e.Message, strings.ToLower(e.ErrorType.String()), e.Entity)
 }
 
+func (e *DomainError) Unwrap() error {
+	return e.Err
+}
+
 func (e *DomainError) DebugString() string {
```

Comment:
need to check on golang has an inbuilt support for this, instead of writing our own.